### PR TITLE
Add Clippy / fmt to CI

### PR DESCRIPTION
**Pull request by dureuill**
# Pull Request

## Related issue

No related issue, maintenance.

### Generally

- Add `cargo clippy` and `cargo fmt` to CI
- Test the `japanese-transliteration` feature in CI
- Fix existing clippy/fmt issues
- Deny warnings in CI

### In details

Each commit of this PR details a change suggested by clippy. The most notable ones are:

- Allowed Cow in shrink_cow, as it looks actually necessary here (clippy false positive)
- Inline classifier::classifier submodule

## PR checklist
Please check if your PR fulfills the following requirements:
- [ ] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Review by ManyTheFish — APPROVED** [review]
It looks good to me, I let @curquiza double check the CI part of the PR and merge it if everything is good!

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -22,3 +24,40 @@ jobs:
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+    - name: Run tests with japanese-transliteration on
+      run: cargo test --verbose --features japanese-transliteration
+
+  clippy:
+    name: Run Clippy
+    runs-on: ubuntu-18.04
```
[the remark this answers is not in the document]
Comment (dureuill) [reply .github/workflows/rust.yml:32]: hmmm, I don't have a strong opinion on this, this should not influence the result (I expect the rust toolchain to always be the latest regardless of the OS).

We might want to change the CI in meilisearch/meilisearch too, then.

[the remark this answers is not in the document]
Comment (dureuill) [reply .github/workflows/rust.yml]: Changed.

**Review by curquiza — APPROVED** [review]

**Comment by ManyTheFish** [comment]
bors merge

**Pull request merged**